include/vara/Feature/ConstraintParser.h

```diff
@@ -497,6 +499,30 @@ class ConstraintParser {
     }
   }
 
+  static std::unique_ptr<Constraint> parseUnsigned(llvm::StringRef Str) {
+    if (Str.contains_lower('e')) {
+      double Number;
+      std::stringstream Stream(Str.str());
+      Stream >> Number;
+      if (Stream.fail()) {
+        llvm::errs() << "Failed to parse number '" << Str << "'\n";
+        return nullptr;
+      }
+      return std::make_unique<PrimaryIntegerConstraint>(round(Number));
+    }
+
+    try {
+      return std::make_unique<PrimaryIntegerConstraint>(std::stol(Str.str()));
+    } catch (std::invalid_argument &_) {
+      llvm::errs() << "Failed to parse number '" << Str << "'\n";
+      return nullptr;
+    } catch (std::out_of_range &_) {
+      llvm::errs() << "Number too large.\n";
+      return std::make_unique<PrimaryIntegerConstraint>(
+          std::numeric_limits<long>::max());
+    }
```

Comment:
Do we have a way to implement this without exception handling? LLVM builds without exceptions and vara-feature is embedded into llvm/vara